lib/matplotlib/colors.py

```diff
@@ -233,8 +233,36 @@ def is_color_like(c):
 
 def _has_alpha_channel(c):
     """Return whether *c* is a color with an alpha channel."""
-    # 4-element sequences are interpreted as r, g, b, a
-    return not isinstance(c, str) and len(c) == 4
+    # If c is not a color, it doesn't have an alpha channel
+    if not is_color_like(c):
+        return False
```

Comment:
One should probably check the code if this has already been called. Since this is a private method one can require that it is checked.

Reply:
Hm, I'm sure I know how to do that or what you mean. Would you explain a little more?  

(also, thanks for the comments, as always)